Feature: SUNDomEigEstimator upgrades (#847)
LLNL/sundials: doc/shared/sundomeigest/SUNDomEigEst_Arnoldi.rst

      :c:func:`SUNDomEigEstimator_SetInitialGuess`.

.. c:function:: SUNErrCode SUNDomEigEstimator_SetRelTol_Arnoldi(SUNDomEigEstimator DEE, sunrealtype rel_tol)

   This routine sets the relative tolerance used during the preprocessing phase
   of the Arnoldi implementation.

   :param DEE: the dominant eigenvalue estimator object.
   :param rel_tol: requested relative tolerance.

   :returns: ``SUN_SUCCESS`` if successful, otherwise an appropriate error code.

   .. note::

      In the Arnoldi implementation, ``rel_tol`` is used only to assess the
      preprocessing Power iterations. Once the preprocessing estimate satisfies

      .. math::

         \left|\lambda_{k} - \lambda_{k-1}\right|
         \le \mathtt{rel\_tol} \cdot |\lambda_{k}|,

      the Arnoldi iteration begins. This avoids restarting Arnoldi repeatedly.

      Supplying ``rel_tol < 0`` disables preprocessing-to-tolerance behavior.
      Inputs satisfying :math:`0 < \mathtt{rel\_tol} < 1 - \varepsilon`
      enable this behavior and are used directly. Values with
      :math:`\mathtt{rel\_tol} = 0` or
      :math:`\mathtt{rel\_tol} > 1 - \varepsilon` reset to the default value

>       ``0.005``.

Reviewer: Including `\varepsilon` is probably more information than the user needs here
```suggestion
      Inputs satisfying :math:`0 < \mathtt{rel\_tol} < 1`
      enable this behavior and are used directly. Values with
      :math:`\mathtt{rel\_tol} = 0` or
      :math:`\mathtt{rel\_tol} >= 1` reset to the default value
      ``0.005``.
```

* ``num_iters`` - number of iterations (preprocessing and estimation) in the
  last :c:func:`SUNDomEigEstimator_Estimate` call,

> * ``warmup_to_tol`` - type of warmup iterations (default is ``SUNFALSE``)

Reviewer 2: ```suggestion
* ``warmup_to_tol`` - enable warmup iterations (default is ``SUNFALSE``)
```

* ``tol_preprocess`` - tolerance for preprocessing iterations (default is 0.005; 
  only used if ``warmup_to_tol`` is ``SUNTRUE``),

* ``rhs_linY`` - state vector for linearization point,

> * ``rhs_linT`` - time value for linearization point (default is 0.0),

Reviewer 2: I don't see how there can be a default `rhs_linT` without a default `rhs_linY`, can we just remove the "default" part of `linT`?
```suggestion
* ``rhs_linY`` - state vector for linearization point,

* ``rhs_linT`` - time value for linearization point,
```

* ``SUNDomEigEstimator_GetNumIters_Arnoldi``

> * ``SUNDomEigEstimator_GetNumRhsCalls_Arnoldi``

Reviewer: ```suggestion
* ``SUNDomEigEstimator_GetNumRhsEvals_Arnoldi``
```
